src/nagini_contracts/thread.py

```diff
@@ -0,0 +1,75 @@
+from typing import Callable, Tuple, Any
+
+
+class Thread:
+    """
+    A stub for threads with a similar interface as threading.Thread.
+    Clients must give the constructor a target method; subclassing and overriding
+    the run method is currently not supported.
+    """
+    def __init__(self, group: object=None, target: Callable=None, name: str=None,
+                 args: Tuple=None, kwargs: object=None, daemon: bool=None) -> None :
+        pass
+
+    def start(self, *method_list : Callable) -> None:
```

Comment:
Yes, start and join are incompatible. This would have to be another thing that's rewritten by the rewriting system. We could make more explicit that these methods don't behave exactly the same way by naming them differently (e.g., start_thread and join_thread)?

Reply:
Perhaps, Marco. Up to you.